### PR TITLE
Reorganization of code and naming

### PR DESCRIPTION
**Pull request by bthaile**
Put logic into internal directory
Consolidated functionality into FlixService
Updated naming

### CODE REVIEW

**Review by chasefleming — CHANGES_REQUESTED** [review]
This is a breaking change. Is there another approach you can take?

**Comment by bthaile** [comment]
> 

Need to return an interface so I can mock it out for unit tests in flow-cli.   I could add new methods that return interfaces and leave the old ones as pass throughs.

**Comment by bthaile** [comment]
> This is a breaking change. Is there another approach you can take?

Updated PR to not have breaking changes. Maintain the old generating binding file methods

File: flixkit/fcl_bindings.go [hunk 1]
```diff
@@ -26,7 +31,12 @@ func NewFclTSGenerator() *FclGenerator {
 	}
 }
 
+// Don't use this one deprecated
 func NewFclJSGenerator() *FclGenerator {
+	return NewFclGeneratorJS().(*FclGenerator)
```
Comment (chasefleming) [inline flixkit/fcl_bindings.go:36]: These names are just the words re-ordered. It's really confusing.

File: flixkit/fcl_bindings.go [hunk 2]
```diff
@@ -104,6 +114,8 @@ type FclGenerator struct {
 	Templates []string
 }
 
+var _ GenerateBinding = (*FclGenerator)(nil)
```
Comment (chasefleming) [inline flixkit/fcl_bindings.go:117]: This naming feels like an action (like a method) as opposed to a struct that that is returned.

**Comment by bthaile** [comment]
It's the same pattern used for `FlixService` it allows mocking out for unit tests. Also better design to return an interface instead of a concrete class. 

Naming is Verb Noun, existing is GenerateTemplate, GenerateBinding. Could start interfaces with `I` but that doesn't seem to be the convention. 

I'm open to whatever naming.

File: flixkit/fcl_bindings.go (hunk 2 above)
Comment (ianthpun) [reply flixkit/fcl_bindings.go:117]: very similiar comment i made for @bthaile privately, we should stay consistent, whichever form we take on

**Comment by bjartek** [comment]
You should consider creating a mock with mockery and publish it. That is what is done in some of the other flow libraries.

**Comment by bthaile** [comment]
> You should consider creating a mock with mockery and publish it. That is what is done in some of the other flow libraries.

Do you have a link to a pattern that you think works well?

**Comment by bjartek** [comment]
Flow cli has some mockery. Just search for it there. I like the expecter options

**Comment by bjartek** [comment]
> Flow cli has some mockery. Just search for it there. I like the expecter options

Sorry for the short comment. On my way! To bed

File: CODEOWNERS [hunk 3]
```diff
@@ -1 +1 @@
-* @chasefleming @bjartek @bthaile
```
Comment (chasefleming) [inline CODEOWNERS:1]: Why is this file in this PR? This was from another commit.

Comment (bthaile) [reply CODEOWNERS:1]: That is weird the file hasn't changed, it's the same as main. 
`* @bjartek @bthaile @ianthpun` is the same as main

**Comment by bthaile** [comment]
Reorganized the code base. Better naming convention.
Added binding file generator to main Flix service, they both need to fetch flix templates. Kept Template generator as it's own since it's functionality is isolated from the rest.

File: internal/flix_service.go [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+Deployed contracts to network addresses
+*/
+type NetworkAddressMap = v1_1.NetworkAddressMap
+
+/*
+contract name associated with network information
+*/
+type ContractInfos = v1_1.ContractInfos
+
+type FlowInteractionTemplateCadence interface {
```
Comment (ianthpun) [inline internal/flix_service.go:57]: does this interface need to be public?

**Pull request merged**